test/delta/test_snapshot.py

```diff
@@ -202,3 +202,51 @@ def fake_open_for_reading(self: SnapshotFile, path: Path) -> SizeLimitedFile:
         with patch("rohmu.delta.snapshot.Path.stat", side_effect=FileNotFoundError):
             with pytest.raises(FileNotFoundError):
                 snapshotter.snapshot(progress=Progress(), reuse_old_snapshotfiles=True)
+
+
+@pytest.mark.timeout(2)
+def test_snapshot_without_create_samples(snapshotter_creator: Callable[..., SnapshotterWithDefaults]) -> None:
+    snapshotter = snapshotter_creator()
+    callback_messages: list[str] = []
+
+    def progress_callback(message: str, success: bool) -> None:
```

Comment:
when we use this function. will it need to be part of class so that it can append the messages to a field in the class?

If so we should define a protocol to ensure that all these methods are part of a class that contains this field

Reply:
This is just an example for the test. progress_callback is used in PGHoard to monitor which phase we are in and whether it has succeeded.